src/mapping.ts

```diff
@@ -1,410 +1,477 @@
-  let underlyingAmount = (option.underlyingAmount as BigInt)
-    .toBigDecimal()
-    .div(
-      exponentToBigDecimal(
-        BigInt.fromI64(
-          ERC20.bind(Address.fromString(underlyingAssetAddress)).decimals()
-        )
-      )
-    );
-
-  let underlyingValueUSD = underlyingPriceUSD
-    .times(underlyingAmount)
-    .times(event.params.amount.toBigDecimal());
-
-  let contract = fetchERC1155(event.address);
-
-  // Update TVL to reflect the value of the exercise tokens being transfered in
-  // and underlying tokens being transfered out
-  contract.totalValueLockedUSD = contract.totalValueLockedUSD
-    .plus(exerciseValueUSD)
-    .minus(underlyingValueUSD);
-
-  contract.save();
-
-  let underlyingAsset = loadOrInitializeToken(
-    Address.fromString(underlyingAssetAddress as string)
+  // handle daily metrics
+  const tokenDaily = fetchDailyTokenMetrics(
+    event.params.asset.toHexString(),
+    event.block.timestamp
   );
-  underlyingAsset.totalValueLocked = underlyingAsset.totalValueLocked.minus(
-    underlyingAmount
+  tokenDaily.notionalVolFeesSwept = tokenDaily.notionalVolFeesSwept.plus(
+    event.params.amount
   );
-  underlyingAsset.totalValueLockedUSD = underlyingAsset.totalValueLockedUSD.minus(
-    underlyingValueUSD
+  tokenDaily.notionalVolFeesSweptUSD = tokenDaily.notionalVolFeesSweptUSD.plus(
+    feeValueUSD
   );
-  underlyingAsset.save();
+  tokenDaily.save();
 
-  let exerciseAsset = loadOrInitializeToken(
-    Address.fromString(exerciseAssetAddress as string)
-  );
-  exerciseAsset.totalValueLocked = exerciseAsset.totalValueLocked.plus(
-    exerciseAmount
-  );
-  exerciseAsset.totalValueLockedUSD = exerciseAsset.totalValueLockedUSD.plus(
-    exerciseValueUSD
+  const dailyOSEMetrics = fetchDailyOSEMetrics(event.block.timestamp);
+  dailyOSEMetrics.notionalVolFeesSweptUSD = dailyOSEMetrics.notionalVolFeesSweptUSD.plus(
+    feeValueUSD
   );
-  exerciseAsset.save();
-
-  let dayData = updateValoremDayData(event);
-
-  dayData.volumeUSD = dayData.volumeUSD.plus(underlyingValueUSD);
+  dailyOSEMetrics.save();
+}
 
-  dayData.save();
+// https://github.com/OpenZeppelin/openzeppelin-subgraphs
+export function handleTransferSingle(event: TransferSingleEvent): void {
+  let contract = fetchERC1155(event.address.toHexString());
+  let operator = fetchAccount(event.params.operator.toHexString());
+  let from = fetchAccount(event.params.from.toHexString());
+  let to = fetchAccount(event.params.to.toHexString());
 
-  let underlyingDayData = updateTokenDayData(underlyingAsset, event);
-  underlyingDayData.volume = underlyingDayData.volume.plus(underlyingAmount);
-  underlyingDayData.volumeUSD = underlyingDayData.volumeUSD.plus(
-    underlyingValueUSD
+  registerTransfer(
+    event,
+    "",
+    contract,
+    operator,
+    from,
+    to,
+    event.params.id,
+    event.params.amount
   );
-  underlyingDayData.save();
 
-  updateTokenDayData(exerciseAsset, event);
+  /**
```

Comment:
I really like how you setup this pattern FWIW with the 0x0 checks and the dedicated function for handling ERC1155 transfer metrics.

Reply:
<3 Thank you